finetune/lora.py

```diff
@@ -50,6 +50,12 @@ def setup(
     out_dir: Path = Path("out/lora/alpaca"),
     precision: Optional[str] = None,
     tpu: bool = False,
+    query_lora: bool = True,
+    key_lora: bool = False,
+    value_lora: bool = True,
+    projection_lora: bool = False,
+    mlp_lora: bool = False,
+    head_lora: bool = False,
```

Comment:
Since lora r, alpha and dropout are configured at the top (requiring code changes) I would keep these together.

Also it would be good to standardize `lora` as a prefix or as a suffix

Reply:
Didn't fully get it.

> Since lora r, alpha and dropout are configured at the top (requiring code changes) I would keep these together

You mean to move r, alpha and dropout inside `setup` function? Or to move these params outside the function?

> Also it would be good to standardize lora as a prefix or as a suffix

Since we have `lora_r`, `lora_alpha` and `lora_dropout`, then `prefix` should be the right choice:
`lora_query`, `lora_key`, ...
But I don't like both options (either prefix or suffix). If to not know what these arguments mean it's pretty hard to understand that it's a boolean value. 
`lora_apply_to_query` is much clearer, but man, it's not far from names of test function/method.
I would like to come up with something short, simple and clear but had no luck with that.

Reply:
> Or to move these params outside the function?

This: to put the new ones as globals alongside the old ones

Reply:
Done. 
By the way what about this:
https://github.com/Lightning-AI/lit-gpt/blob/0215307262150dcc59572f8b7868bf6ac33cd042/lit_gpt/lora.py#L462-L464

shouldn't these params also have `lora_` prefix?

Reply:
I would say that none should have it in the lora specific config.

> I would like to come up with something short, simple and clear but had no luck with that.

My suggestion is `to_key, to_query`, ...

Reply:
Like it.
Changed.
Thanks.
![image](https://github.com/Lightning-AI/lit-gpt/assets/58434077/283bc4b8-41e3-4385-bc9c-8e32e4f4b6a8)